### PR TITLE
Update ci.yml

### PR DESCRIPTION
**Pull request by lhstrh**
CI gets a bit overwhelmed if we build for every push. Let's limit to PRs and pushes to main.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -2,6 +2,8 @@ name: CI
 
 on:
   push:
+    branches:
+      - master
```
Comment (Soroosh129) [inline .github/workflows/ci.yml:6]: Should this be `main`?

Comment (lhstrh) [reply .github/workflows/ci.yml:6]: Ah, right. Nice catch!

Comment (lhstrh) [inline .github/workflows/ci.yml:6]:
```suggestion
      - main
```

**Review by Soroosh129 — APPROVED** [review]

**Comment by Soroosh129** [comment]
Looks good to me! Thanks for updating this.

**Pull request merged**